.tool-versions

```diff
@@ -0,0 +1 @@
+java zulu-11.41.23
```

Comment:
whats that for? I don't know what a `.tool-versions` does.
Why is java a required tool? What is zulu?

Reply:
`.tools.versions` comes from [asdf version manager](https://asdf-vm.com#/) , a tool that helps creating reproducible development environments on Linux and Mac.
If you do not use it, it has no effect, except, if gives documentation about used tools/version

Java is there, because on non Android platforms you have a dependency on JNI, for Java bindings.
[zulu jdk](https://www.azul.com/downloads/zulu-community/?architecture=x86-64-bit&package=jdk) is, like [adoptopenjdk](https://adoptopenjdk.net) or [openjdk](https://openjdk.java.net), a Java distribution.

The beauty of asdf is, you can toggle between those as you wish.
I use asdf also for managing my scala, sbt, ruby, python cmake and some other installations. I heavily recommend the usage of it for every developer. You can, for example, test the generator with different java versions super easy.
And once we have downloads of djinni generator, I plan to add a asdf plugin to get djinni, because that's an easy and convenient way to get it on you system

Reply:
Ok, now I get it. For the build to succeed a jdk needs to be installed for the jni-headers to be available. Thats essential for pre-building the library for distribution. 👍